gsl_simulation/views.py

```diff
@@ -41,13 +48,100 @@ def get_queryset(self):
         visible_by_user_enveloppes = EnveloppeService.get_enveloppes_visible_for_a_user(
             self.request.user
         )
-        return Simulation.objects.filter(
+        qs = Simulation.objects.filter(
             enveloppe__in=visible_by_user_enveloppes
         ).order_by("-created_at")
+        qs = qs.select_related(
+            "enveloppe",
+            "enveloppe__perimetre",
+            "enveloppe__perimetre__region",
+            "enveloppe__perimetre__departement",
+        )
+
+        return qs
+
+
+class SimulationProjetListViewFilters(ProjetFilters):
+    filterset = (
+        "porteur",
+        "status",
+        "cout_total",
+        "montant_demande",
+        "montant_previsionnel",
+    )
+
+    ordered_status = (
+        SimulationProjet.STATUS_DRAFT,
+        SimulationProjet.STATUS_PROVISOIRE,
+        SimulationProjet.STATUS_CANCELLED,
+        SimulationProjet.STATUS_VALID,
+    )
+
+    status = MultipleChoiceFilter(
+        field_name="simulationprojet__status",
+        choices=order_couples_tuple_by_first_value(
+            SimulationProjet.STATUS_CHOICES, ordered_status
+        ),
+        widget=CustomCheckboxSelectMultiple(),
+        method="filter_status",
+    )
+
+    def filter_status(self, queryset, name, value):
+        return queryset.filter(
+            # Cette ligne est utile pour qu'on ait un "ET", cad, on filtre les projets de la simulation en cours ET sur les statuts sélectionnés.
+            # Sans ça, on aurait dans l'ordre :
+            # - les projets dont IL EXISTE UN SIMULATION_PROJET (pas forcément celui de la simulation en question) qui a un des statuts sélectionnés
+            # - les simulation_projets de la simulation associés aux projets filtrés
+            simulationprojet__simulation__slug=self.request.resolver_match.kwargs.get(
+                "slug"
+            ),
+            simulationprojet__status__in=value,
```

Comment:
Pour dé-dupliquer cette logique on peut imaginer quelque chose comme ça : 
```suggestion
            simulationprojet__status__in=value,
            **self.simulation_slug_filter_kwarg()
```

où la méthode `simulation_slug_filter_kwarg` retournerait un dict : 

```python
def simulation_slug_filter_kwarg():
    return {
            "simulationprojet__simulation__slug": self.request.resolver_match.kwargs.get("slug")
        }
```

Reply:
Éventuellement en renommant la méthode "common_filter_kwargs" si on se rend compte qu'on a d'autres kwargs dont on a besoin un peu tout le temps ?